src/Development/Shake/FilePattern.hs

```diff
@@ -184,7 +184,7 @@ compatible (x:xs) = all ((==) (specials x) . specials) xs
 
 -- | Extract the items that match the wildcards. The pair must match with '?=='.
 extract :: FilePattern -> FilePath -> [String]
-extract p@(parse -> pat) = \x -> case match pat (split isPathSeparator x) of
+extract p@(parse -> pat) x = case match pat (split isPathSeparator x) of
```

Comment:
See https://github.com/ndmitchell/blogs/blob/master/hlint-view-patterns.md - it was confusing to me too, so I benchmarked on the train and wrote a blog post (that's the draft version, I'll finish it off tonight with any luck).